bodo/tests/test_df_lib/test_end_to_end.py

```diff
@@ -682,34 +686,36 @@ def test_head(datapath):
 
 def test_apply(datapath, index_val):
     """Very simple test for df.apply() for sanity checking."""
-    df = pd.DataFrame(
-        {
-            "a": pd.array([1, 2, 3] * 10, "Int64"),
-            "b": pd.array([4, 5, 6] * 10, "Int64"),
-            "c": ["a", "b", "c"] * 10,
-        },
-        index=index_val[:30],
-    )
-    bdf = bd.from_pandas(df)
-    out_pd = df.apply(lambda x: x["a"] + 1, axis=1)
-    out_bodo = bdf.apply(lambda x: x["a"] + 1, axis=1)
+    with assert_executed_plan_count(1):
+        df = pd.DataFrame(
+            {
+                "a": pd.array([1, 2, 3] * 10, "Int64"),
+                "b": pd.array([4, 5, 6] * 10, "Int64"),
+                "c": ["a", "b", "c"] * 10,
+            },
+            index=index_val[:30],
+        )
+        bdf = bd.from_pandas(df)
+        out_pd = df.apply(lambda x: x["a"] + 1, axis=1)
+        out_bodo = bdf.apply(lambda x: x["a"] + 1, axis=1)
     _test_equal(out_bodo, out_pd, check_pandas_types=False)
 
 
 def test_chain_python_func(datapath, index_val):
     """Make sure chaining multiple Series functions that run in Python works"""
-    df = pd.DataFrame(
-        {
-            "A": pd.array([1, 2, 3, 7], "Int64"),
-            "B": ["A1\t", "B1 ", "C1\n", "Abc\t"],
-            "C": pd.array([4, 5, 6, -1], "Int64"),
-        }
-    )
-    df.index = index_val[: len(df)]
-    bdf = bd.from_pandas(df)
-    out_pd = df.B.str.strip().str.lower()
-    out_bodo = bdf.B.str.strip().str.lower()
-    assert out_bodo.is_lazy_plan()
+    with assert_executed_plan_count(0):
+        df = pd.DataFrame(
+            {
+                "A": pd.array([1, 2, 3, 7], "Int64"),
+                "B": ["A1\t", "B1 ", "C1\n", "Abc\t"],
+                "C": pd.array([4, 5, 6, -1], "Int64"),
+            }
+        )
+        df.index = index_val[: len(df)]
+        bdf = bd.from_pandas(df)
+        out_pd = df.B.str.strip().str.lower()
+        out_bodo = bdf.B.str.strip().str.lower()
+        assert out_bodo.is_lazy_plan()
```

Comment:
Maybe in general we can remove these is_lazy_plan as that seems redundant with the count check.